src/uct/api/uct_def.h

```diff
@@ -99,7 +99,13 @@ typedef int                      uct_worker_cb_id_t;
  * @brief Structure for scatter-gather I/O.
  *
  * Specifies a list of buffers which can be used within a single data transfer
- * function call.
+ * function call, or to be registered as a memory layout with @ref uct_ep_mem_reg .
+ *
+ * A single entry of this data structure represents either a contiguous array of
+ * items or a simple strided vector of items. An X-dimensional strided vector would
+ * consist of X consecutive structures, where only the first has a non-zero buffer
+ * pointer, and each contains the stride and count of the i-th dimension.
```

Comment:
seems counter-intuitive to require the user to pass a "serialized tree". Why not use the memory key to describe a hierarchy?